aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1192,7 +1192,40 @@ void magmaGels<c10::complex<double>>(
       reinterpret_cast<magmaDoubleComplex*>(hwork), lwork, info);
   AT_CUDA_CHECK(cudaGetLastError());
 }
-#endif
+
+namespace {
+
+/*
+  MAGMA can return errors both as a return value and in the info argument.
+  The return value and info should always be identical.
+  In general, the meaning is as given in this table.
+  Predefined error codes are large negative numbers. Using the symbolic
+  constants below is preferred, but the numeric values can be found in
+  include/magma_types.h.
+
+  Info                       |  Description
+  -----------                |  -----------
+  info = 0 (MAGMA_SUCCESS)   |  Successful exit
+  info < 0, but small        |  For info = -i, the i-th argument had an illegal value
+  info > 0                   |  Function-specific error such as singular matrix
+  MAGMA_ERR_DEVICE_ALLOC     |  Could not allocate GPU device memory
+  MAGMA_ERR_HOST_ALLOC       |  Could not allocate CPU host memory
+  MAGMA_ERR_ILLEGAL_VALUE    |  An argument had an illegal value (deprecated; instead it should return -i to say the i-th argument was bad)
+  MAGMA_ERR_INVALID_PTR      |  Can't free pointer
+  MAGMA_ERR_NOT_IMPLEMENTED  |  Function or option not implemented
+  MAGMA_ERR_NOT_SUPPORTED    |  Function or option not supported on the current architecture
+*/
+void checkMagmaInternalError(magma_int_t info, const std::string& magma_function_name) {
+  // if info > 0 the error is function-specific, do nothing in this case
+  TORCH_CHECK(info >= 0,
```

Comment:
What's an example for the value of MAGMA_ERR_NOT_IMPLEMENTED?

Also won't this check trigger both when the ith argument has an illegal value and when a MAGMA-specific error code is returned?

Reply:
I think we don't use any function from MAGMA that could give MAGMA_ERR_NOT_IMPLEMENTED.

Yes, this check would trigger both when the ith argument has an illegal value and when a MAGMA-specific error code is returned. If two problems are present what is actually returned depends on magma's source code. This error check doesn't cause device synchronization because the `magma_int_t info` is always on host (CPU) memory.

Reply:
`magma_strerror` gives a string both for internal failures and invalid arguments.
See https://bitbucket.org/icl/magma/src/4c2588119ee636999b230b248fca3e3824a23be9/interface_cuda/error.cpp?at=master#lines-137

Reply:
I added printing the actual value of `info` to the error message so that we know which argument to the call is invalid. For other error types, the value is meaningless.